src/firefly/java/edu/caltech/ipac/firefly/server/util/VersionUtil.java

```diff
@@ -120,6 +127,70 @@ private static String getFireflyUA() {
             return unknownVersionUA;
         }
     }
+
+    public static List<KeyVal<String, String>> getVersionInfo() {
+        Version v = getAppVersion();
+
+        List<KeyVal<String, String>> versionInfo = new ArrayList<>(6);
+        versionInfo.add(new KeyVal<>("Version", getVersionStr(v)));
+        versionInfo.add(new KeyVal<>("Built On", v.getBuildTime()));
+        versionInfo.add(new KeyVal<>("Git commit", v.getBuildCommit()));
+
+        if (v.getMajor()>0)
+            versionInfo.add(new KeyVal<>("Firefly Library Version", getFireflyVersionStr(v)));
+        if (!StringUtils.isEmpty(v.getBuildCommitFirefly()))
+            versionInfo.add(new KeyVal<>("Firefly Git Commit", v.getBuildCommitFirefly()));
+        if (!StringUtils.isEmpty(v.getBuildFireflyTag()))
+            versionInfo.add(new KeyVal<>("Git commit", v.getBuildCommit()));
+
+        return versionInfo;
+    }
+
+    private static String getVersionStr(Version v) {
+        Integer major = v.getMajor();
+
+        if (major>0) {
+            String versionStr = String.format("v%s.%s", major, v.getMinor());
+            String buildRev = v.getRev();
+            versionStr += buildRev.equals("0") ? "" : String.format(".%s", buildRev);
+            return versionStr;
+        }
+        else return getFireflyVersionStr(v);
+    }
+
+    private static String getFireflyVersionStr(Version v) {
+        String fireflyTag = v.getBuildFireflyTag();
+        String fireflyCommit = v.getBuildCommitFirefly();
+        String fireflyBranch = v.getBuildFireflyBranch();
+        if (StringUtils.isEmpty(fireflyBranch)) fireflyBranch = "unknown-branch";
+
+        Matcher formalReleaseMatcher = Pattern.compile("release-(\\d+(\\.\\d+)+)").matcher(fireflyTag);
+        if (formalReleaseMatcher.find()) return formalReleaseMatcher.group(1);
```

Comment:
You are not trying to validate `fireflyTag`.  You just want to extract the info.  This is a lot easier to read, no?

```java
if (fireflyTag.startsWith("release-")    return fireflyTag.substring(8);
```

Reply:
This is definitely easier to read. 

But I thought there might be a case where version is "release-some-non-version-string" so we beed to validate using regex. I assumed that based onthe JS code that does all of this by regex matching: https://github.com/Caltech-IPAC/firefly/blob/dev/src/firefly/js/Firefly.js#L269-L297

Let me know if that's not the case. I will change it to what you suggested.

Reply:
This is our tag naming convention.  It is not an input.  I can't imagine why anyone would want to create a 'release-' tag that's not for a release.  But, it's a legitimate reason if you want to keep the logic the same as Firefly.js.  I am ok with that.